core/state/statedb.go

```diff
@@ -674,6 +679,15 @@ func (db *StateDB) ForEachStorage(addr common.Address, cb func(key, value common
 	return nil
 }
 
+// copyPredicateStorageSlots creates a deep copy of the provided predicateStorageSlots map.
+func copyPredicateStorageSlots(predicateStorageSlots map[common.Address][]common.Hash) map[common.Address][]common.Hash {
+	res := make(map[common.Address][]common.Hash)
+	for address, slots := range predicateStorageSlots {
+		res[address] = slots
+	}
```

Comment:
should we do a deep copy of the hashes?

Reply:
As per the comment above, I think we should change the value type in this map to a byte slice and then agree on copying the bytes here.

Reply:
Fixed this to make a deep copy.